### ISSUE TITLE
 S3216 FP: When used in code with thread affinity

### ISSUE THREAD

**Issue by dmitry-golovinov-sonarsource** [SonarSource/sonar-dotnet#4522]
### Description

S3216 triggers a FP, when the code after await has a thread affinity. (most common example - WPF and Windows Forms controls)

### Repro steps

```
private async void Button_Clicked(object sender, RoutedEventArgs e)
{
	await Task.Delay(1); // FP
	Random r = new Random();
	Background = new SolidColorBrush(Color.FromRgb(
		(byte)r.Next(255),
		(byte)r.Next(255),
		(byte)r.Next(255)));
}

```
### Expected behavior

Issue should not be raised, since the following code should run in the same thread.

### Actual behavior

Issue is being raised.

### Known workarounds

Disable rule completely or with the help of editorconfig

### Related information

* C#/VB.NET Plugins version 8.22

**Comment by jacksonantbhf** [comment]
+1 - my team has fallen foul for this rule and started added ConfigureAwait(false) everywhere. If it hadn't been spotted quite quickly this could have led to strange runtime bugs due to code executing without the expected thread context. 

The very clear recommendation from MS is not to use ConfgureAwait(false) in application code. Presumably this accounts for most code you are scanning and for that reason you should remove this rule. See here for MS advice: 

https://devblogs.microsoft.com/dotnet/configureawait-faq/#when-should-i-use-configureawaitfalse

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1907](https://sonarsource.atlassian.net/browse/NET-1907)

[NET-1907]: https://sonarsource.atlassian.net/browse/NET-1907?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
